drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLBlockVisitor.java

```diff
@@ -132,17 +132,7 @@ public Block enter(Block block) {
 
         if (block.isLoopHeader()) {
             loopCount++;
-            openclBuilder.emitLoopHeader(block);
-            // Temporary fix to remove the end scope of the most outer loop
-            // without changing the loop schematics in IR level.
-            if (openclBuilder.shouldRemoveLoop()) {
```

Comment:
@gigiblender , did you check on FPGAs? We have this condition for optimizing some type of code on FPGAs.

Reply:
Yes, the outer most `@Parallel` loop is not generated on FPGAs

Reply:
https://github.com/beehive-lab/TornadoVM/blob/202a27b0ee1f58368feab95619d9e108e7b52cda/drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLCompilationResultBuilder.java#L263
This condition prevents it from being generated